.github/workflows/linux-build-base.yml

```diff
@@ -18,45 +18,61 @@ on:
   workflow_call:
     inputs:
       use-clang:
-        description: 'Use Clang to compile the project.'
+        description: Use Clang to compile the project.
         default: false
         required: false
         type: boolean
 
 jobs:
   ubuntu-release:
-    name: "Ubuntu release with resolve_dependency"
+    name: Ubuntu release with resolve_dependency
     runs-on: yscope-gh-runner
     env:
-      CCACHE_DIR: "${{ github.workspace }}/ccache"
-      USE_CLANG: "${{ inputs.use-clang && 'true' || 'false' }}"
+      CCACHE_CONFIGPATH: ${{ github.workspace }}/ccache.conf
+      CCACHE_DIR: ${{ github.workspace }}/ccache
+      USE_CLANG: ${{ inputs.use-clang && 'true' || 'false' }}
     defaults:
       run:
         shell: bash
         working-directory: velox
     steps:
-      - name: Ensure Stash Dirs Exists
-        working-directory: ${{ github.workspace }}
-        run: |
-          mkdir -p '${{ env.CCACHE_DIR }}'
-
       - uses: actions/checkout@v4
         with:
           path: velox
+          persist-credentials: false
+
+      - name: Get Release Build Ccache Stash
+        run: |
+          rm -rf "$CCACHE_DIR"
+          cp -r ~/yscache/velox-ubuntu-release-"${{ inputs.use-clang && 'clang' || 'gcc' }}"/ccache "$CCACHE_DIR"
+          cp ~/yscache/velox-ubuntu-release-"${{ inputs.use-clang && 'clang' || 'gcc' }}"/ccache.conf "$CCACHE_CONFIGPATH"
```

Comment:
Instead of moving these directories around, we can just let ccache fully manage the cache:

* ccache already handles keeping track of which compiler a file was built with, so we don't need to do that ourselves.
* I guess the reason you want to only "commit" the cache is if the build is successful. However, let's say a source file gets cached even though the build fails; that cache entry will either get reused in the future or eventually get evicted because it's not used frequently, right?